astropy/modeling/tests/test_model_sets.py

```diff
@@ -220,5 +214,39 @@ def test_linearlsqfitter():
 
     m1 = Polynomial1D(1, c0=fit.c0[0][0], c1=fit.c1[0][0])
     m2 = Polynomial1D(1, c0=fit.c0[0][1], c1=fit.c1[0][1])
-    assert_allclose(model_y[0], m1(x))
-    assert_allclose(model_y[1], m2(x))
+    assert_allclose(model_y[:, 0], m1(x))
+    assert_allclose(model_y[:, 1], m2(x))
+
+
+def test_model_set_axis_outputs():
+    fitter = LinearLSQFitter()
+    model_set = Polynomial2D(1, n_models=2, model_set_axis=2)
+    y2, x2 = np.mgrid[: 5, : 5]
+    # z = np.moveaxis([x2 + y2, 1 - 0.1 * x2 + 0.2 * y2]), 0, 2)
+    z = np.rollaxis(np.array([x2 + y2, 1 - 0.1 * x2 + 0.2 * y2]), 0, 3)
+    model = fitter(model_set, x2, y2, z)
+    res = model(x2, y2, model_set_axis=False)
+    assert z.shape == res.shape
+
+    # Test initializing with integer model_set_axis
+    # and evaluating with a different model_set_axis
+    model_set = Polynomial1D(1, c0=[1, 2], c1=[2, 3],
+                             n_models=2, model_set_axis=0)
+    y0 = model_set(xx)
+    y1 = model_set(xx.T, model_set_axis=1)
+    assert_allclose(y0[0], y1[:, 0])
+    assert_allclose(y0[1], y1[:, 1])
+
+    model_set = Polynomial1D(1, c0=[[1, 2]], c1=[[2, 3]],
+                             n_models=2, model_set_axis=1)
+    y0 = model_set(xx.T)
+    y1 = model_set(xx, model_set_axis=0)
+    assert_allclose(y0[:, 0], y1[0])
+    assert_allclose(y0[:, 1], y1[1])
+    with pytest.raises(ValueError):
+        model_set(x)
+
+    y0 = model_set(xx.T)
```

Comment:
Sorry I wasn't clear enough. I think this whole block from here to the end is also a duplicate?

Reply:
No, it's me. I'm blind. Rebased and cleaned now.